src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp

```diff
@@ -1645,6 +1645,32 @@ namespace {
         }
     }
 
+    void Schedule::handleWSEGAICD(const HandlerContext& handlerContext, const ParseContext&, ErrorGuard&) {
+        std::map<std::string, std::vector<std::pair<int, AutoICD> > > auto_icds = AutoICD::fromWSEGAICD(handlerContext.keyword);
+
+        for (auto& map_elem : auto_icds) {
```

Comment:
In situations like this I have become very found of form:
```
for (auto& [well_name_pattern, aicd_pairs] : auto_icds) {
     ....
}
``` 
to avoid accessing the `.first` and `.second` members. Just a tip.